src/main/scala/io/qbeast/spark/delta/IndexFiles.scala

```diff
@@ -0,0 +1,199 @@
+   * @param dataChange
+   *   the index file represents a data change
+   * @param indexFile
+   *   the IndexFile instance
+   * @return
+   *   an AddFile instance
+   */
+  def toAddFile(dataChange: Boolean = true)(indexFile: IndexFile): AddFile = {
+    val tags = Map(
+      TagUtils.revision -> indexFile.revisionId.toString,
+      TagUtils.blocks -> encodeBlocks(indexFile.blocks))
+    AddFile(
+      path = indexFile.path,
+      partitionValues = Map.empty[String, String],
+      size = indexFile.size,
+      modificationTime = indexFile.modificationTime,
+      dataChange = dataChange,
+      tags = tags)
+  }
+
+  /**
+   * Converts a given IndexFile instance to a RemoveFile instance.
+   *
+   * @param dataChange
+   *   file removal implies data change
+   * @param indexFile
+   *   the IndexFile instance
+   * @param a
+   *   RemoveFile instance
+   */
+  def toRemoveFile(dataChange: Boolean = false)(indexFile: IndexFile): RemoveFile =
+    RemoveFile(
+      path = indexFile.path,
+      deletionTimestamp = Some(System.currentTimeMillis()),
+      dataChange = dataChange,
+      partitionValues = Map.empty[String, String],
+      size = Some(indexFile.size))
+
+  /**
+   * Converts IndexFile instance to FileStatus instance.
+   *
+   * @param indexPath
+   *   the path to the root of the index
+   * @param indexFile
+   *   the IndexFile instance
+   * @return
+   *   a FileStatus instance
+   */
+  def toFileStatus(indexPath: Path)(indexFile: IndexFile): FileStatus = {
+    var path = new Path(new URI(indexFile.path))
+    if (!path.isAbsolute()) {
+      path = new Path(indexPath, path)
+    }
+    new FileStatus(indexFile.size, false, 0, 0, indexFile.modificationTime, path)
+  }
+
+  def toFileStatusWithMetadata(indexPath: Path, metadata: Map[String, String] = Map.empty)(
+      indexFile: IndexFile): FileStatusWithMetadata = {
+    FileStatusWithMetadata(toFileStatus(indexPath)(indexFile), metadata)
+  }
+
+  private def encodeBlocks(blocks: IISeq[Block]): String = {
+    val writer = new StringWriter()
+    val generator = jsonFactory.createGenerator(writer)
+    generator.writeStartArray()
+    blocks.foreach { block =>
+      generator.writeStartObject()
+      generator.writeStringField("cubeId", block.cubeId.string)
+      generator.writeNumberField("minWeight", block.minWeight.value)
+      generator.writeNumberField("maxWeight", block.maxWeight.value)
+      generator.writeNumberField("elementCount", block.elementCount)
+      generator.writeBooleanField("replicated", block.replicated)
+      generator.writeEndObject()
+    }
+    generator.writeEndArray()
+    generator.close()
+    writer.close()
+    writer.toString()
+  }
+
+  private def decodeBlocks(json: String, dimensionCount: Int, builder: IndexFileBuilder): Unit = {
+    val parser = jsonFactory.createParser(json)
+    parser.nextToken()
+    if (!parser.hasToken(JsonToken.START_ARRAY)) {
+      throw new JsonParseException(parser, "Array start is expected")
```

Comment:
Same issue in this file about not having tests that validate that the code throws as expected.

Reply:
I would add some test for IndexFiles. Seems there's none specific for the object.

Reply:
Issue #296 opened.